test/e2e/v1beta1/scripts/gh-actions/setup-katib.sh

```diff
@@ -65,8 +65,27 @@ echo "Deploying Katib"
 cd ../../../../../ && WITH_DATABASE_TYPE=$WITH_DATABASE_TYPE make deploy && cd -
 
 # Wait until all Katib pods is running.
+MAX_RETRY_COUNT=24
+COMPONENTS_NUM=3
+if "${DEPLOY_KATIB_UI}"; then
+  COMPONENTS_NUM=4
+fi
+until [ "$(kubectl get pods -n kubeflow -l "katib.kubeflow.org/component in ($WITH_DATABASE_TYPE,controller,ui,db-manager)" -oname | wc -l | tr -d ' ')" = "${COMPONENTS_NUM}" ]; do
+  echo -e "\nWaiting for katib components to be created"
+  kubectl get pods -n kubeflow || :
+  kubectl describe replicasets -n kubeflow || :
+
+  sleep 5
+  MAX_RETRY_COUNT=$((MAX_RETRY_COUNT - 1))
+  if [ "${MAX_RETRY_COUNT}" = "0" ]; then
+    echo "Failed to create katib components"
+    exit 1
+  fi
+done
+
 TIMEOUT=120s
-kubectl wait --for=condition=ready --timeout=${TIMEOUT} -l "katib.kubeflow.org/component in ($WITH_DATABASE_TYPE,controller,db-manager,ui)" -n kubeflow pod ||
+
+kubectl wait --for=condition=ContainersReady=True --timeout=${TIMEOUT} -l "katib.kubeflow.org/component in ($WITH_DATABASE_TYPE,controller,db-manager,ui)" -n kubeflow pod ||
```

Comment:
The `--for=condition=ready` option doesn't confirm if all containers are ready:

```shell
/home/runner/work/katib/katib/test/e2e/v1beta1/scripts/gh-actions
pod/katib-db-manager-756ff4d844-wrs27 condition met
All Katib components are running.
Katib deployments
NAME               READY   UP-TO-DATE   AVAILABLE   AGE
katib-controller   0/1     0            0           2s
katib-db-manager   1/1     1            1           2s
katib-mysql        0/1     0            0           2s
```

https://github.com/kubeflow/katib/actions/runs/5842762640/job/15844213190?pr=2201#step:4:5605